api/src/lib/webhonc/index.ts

```diff
@@ -0,0 +1,161 @@
+  );
+
+  const handler = messageHandlers[parsedMessage.event] as (
+    message: typeof parsedMessage,
+    wsConnections: Set<WebSocket>,
+    db: LibSQLDatabase<typeof schema>,
+  ) => Promise<void>;
+
+  if (handler) {
+    await handler(parsedMessage, wsConnections, db);
+  } else {
+    logger.error(`Unhandled event type: ${parsedMessage.event}`);
+  }
+}
+
+const messageHandlers: {
+  [K in z.infer<typeof WsMessageSchema>["event"]]: (
+    message: Extract<z.infer<typeof WsMessageSchema>, { event: K }>,
+    wsConnections: Set<WebSocket>,
+    db: LibSQLDatabase<typeof schema>,
+  ) => Promise<void>;
+} = {
+  trace_created: async (_message, _wsConnections, _db) => {
+    logger.info("trace_created message received, no action required");
+  },
+  connection_open: async (message, wsConnections) => {
+    const { connectionId } = message.payload;
+    setWebHoncConnectionId(connectionId);
+    for (const ws of wsConnections) {
+      ws.send(
+        JSON.stringify({
+          event: "connection_open",
+          payload: { connectionId },
+        }),
+      );
+    }
+  },
+  request_incoming: async (message, _wsConnections, db) => {
+    // no trace id is coming from the websocket, so we generate one
+    const traceId = crypto.randomUUID();
+
+    const serviceTarget = resolveServiceArg(process.env.FPX_SERVICE_TARGET);
+    const resolvedPath = path.join(serviceTarget, ...message.payload.path);
+    const requestUrl = resolveUrlQueryParams(
+      resolvedPath,
+      message.payload.query,
+    );
+
+    const startTime = Date.now();
+
+    const newRequest: schema.NewAppRequest = {
+      requestMethod: message.payload
+        .method as schema.NewAppRequest["requestMethod"],
+      requestUrl,
+      requestHeaders: message.payload.headers,
+      requestPathParams: {},
+      requestQueryParams: message.payload.query,
+      requestBody: message.payload.body,
+      requestRoute: message.payload.path.join("/"),
+    };
+
+    const webhoncId = getWebHoncConnectionId();
+
+    const supplementedHeaders = {
+      "x-fpx-trace-id": traceId,
+      "x-fpx-webhonc-id": webhoncId ?? "",
+    };
+
+    if (newRequest?.requestHeaders) {
+      newRequest.requestHeaders = {
+        ...newRequest?.requestHeaders,
+        ...supplementedHeaders,
+      };
+    } else {
+      newRequest.requestHeaders = supplementedHeaders;
+    }
+
+    const [{ id: requestId }] = await db
+      .insert(schema.appRequests)
+      .values(newRequest)
+      .returning({ id: schema.appRequests.id });
+
+    try {
+      const response = await executeProxyRequest({
+        requestHeaders: message.payload.headers,
+        // @ts-expect-error - Trust me, the request method is correct, and it's a string
+        requestMethod: message.payload.method,
+        requestBody: message.payload.body,
+        requestUrl,
+      });
+
+      const duration = Date.now() - startTime;
+
+      await handleSuccessfulRequest(db, requestId, duration, response);
```

Comment:
note - this will need to accept the traceId from above, since the fpx otel library no longer returns the traceId in its headers ((i tried to do that and failed, it's harder now that we are no longer middleware))

Reply:
that change is already on my otel branch though, so don't worry about making it here